lib/AST/Decl.cpp

```diff
@@ -2183,6 +2183,29 @@ bool ValueDecl::isProtocolRequirement() const {
   return true;
 }
 
+bool ValueDecl::isDefaultImplInProtocol() const {
+  if (isa<ProtocolDecl>(getDeclContext())) {
+    if (auto *func = dyn_cast<AbstractFunctionDecl>(this)) {
+      if (func->hasBody()) {
+        return true;
+      }
+    }
+    
+    if (auto *storage = dyn_cast<AbstractStorageDecl>(this)) {
+      if (storage->hasAnyAccessors()) {
+        auto accessors = storage->getAllAccessors();
+        
+        for (auto *accessor : accessors) {
+          if (accessor->hasBody())
```

Comment:
I initially was checking if all accessors had bodies, but I ran into a problem where the implicit modify coroutine accessor did not have a body. The modify accessor was only synthesizing it’s body after the storage decl was added to name lookup. I wanted to avoid force triggering a synthesis of the modify accessor.

Reply:
In that case you might want to be more explicit here about what invariants you expect. CC @rjmccall who might have some comments.

Reply:
Yeah, there's some code around there that definitely interprets "defined in a protocol" with "is totally abstract".